File: app/src/main/java/com/stephenbain/obliquestrategies/di/AppComponent.java
package com.stephenbain.obliquestrategies.di;

import com.stephenbain.obliquestrategies.view.HomeActivity;
import dagger.Component;

import javax.inject.Singleton;

@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {
    void inject(HomeActivity activity);
}

File: app/src/main/java/com/stephenbain/obliquestrategies/di/AppComponent.kt
package com.stephenbain.obliquestrategies.di

import com.stephenbain.obliquestrategies.MyApplication
import com.stephenbain.obliquestrategies.view.HomeActivity
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class))
interface AppComponent {
    fun inject(application: MyApplication)
    fun inject(activity: HomeActivity)
}